File: frontend/webapp/src/features/catalog/models/Product.ts
export interface ProductResponse {
    id: number;
    name: string;
    description: string;
    brand: string;
    richDescription: string | null;
    isRichDescriptionAIGenerated: boolean;
    basePrice: number;
    isActive: boolean;
    createdAt: string;
    primaryImageUrl: string | null;
    categoryId: number;
    categoryName: string | null;
    images: ProductImageResponse[];
    variants: ProductVariantResponse[];
    tags: string[];
    aiStatus: string[];
}

export interface ProductImageResponse {
    url: string;
    isPrimary: boolean;
}

export interface ProductVariantResponse {
    sku: string;
    price: number;
    stockQuantity: number;
    imageUrl: string | null;
    color: string | null;
    size: string | null;
    material: string | null;
}

export interface ProductUpsertRequest {
    name: string;
    description: string;
    brand: string;
    richDescription: string | null;
    useAIGeneratedRichDescription: boolean;
    isActive: boolean;
    categoryId: number;
    images: ProductImageRequest[];
    variants: ProductVariantRequest[];
}

export interface ProductImageRequest {
    url: string;
    isPrimary: boolean;
}

export interface ProductVariantRequest {
    sku: string;
    price: number;
    stockQuantity: number;
    imageUrl: string | null;
    color: string | null;
    size: string | null;
    material: string | null;
}

File: services/Catalog/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Catalog.Models;

public class Product
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Brand { get; set; } = string.Empty;
    public string? RichDescription { get; set; }
    public bool IsRichDescriptionAIGenerated { get; set; } = false;

    public List<ProductImage> Images { get; set; } = new();

    public string? PrimaryImageUrl => Images.FirstOrDefault(i => i.IsPrimary)?.Url
                                      ?? Images.FirstOrDefault()?.Url;

    public int CategoryId { get; set; }
    public ProductCategory? Category { get; set; }
    public List<ProductTag> Tags { get; set; } = new();

    public decimal BasePrice { get; set; }

    [NotMapped]
    [Obsolete("Use BasePrice instead.")]
    public decimal Price
    {
        get => BasePrice;
        set => BasePrice = value;
    }

    [NotMapped]
    public bool UseAIGeneratedRichDescription { get; set; }

    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public List<ProductVariant> Variants { get; set; } = new();
}
